andi/andi.py

```diff
@@ -7,11 +9,40 @@
 from andi.typeutils import get_union_args, is_union
 
 
+def _get_globalns_as_get_type_hints(func) -> Dict:
+    """ Global namespace resolution extracted from ``get_type_hints`` method """
```

Comment:
Where is this function copied from? It'd be nice to have a link here, maybe to cpython repo on github.

Could you please also check https://codecov.io/gh/scrapinghub/andi/compare/b64bf74d4182e3a96563c4e35ec10d396d2164a4...665c6a53bb5ba511dfbfcbe25223d915c91500e8/diff? It shows that parts of this code are never called in tests.

I'm asking because locally for me the relevant code is just 

```py
    if globalns is None:
        globalns = getattr(obj, '__globals__', {})
```

This is likely because it is from Python 3.6 - what Python version have you copied it from? It would be good to preserve this, as the code is changing upstream.

Reply:
I copied from 3.7: https://github.com/python/cpython/blob/3.7/Lib/typing.py#L981-L988 . I've put the link in the docstring here:https://github.com/scrapinghub/andi/pull/2/commits/3cf30d3a82f0f74189bc6195f8ea674c0c7ba361

> It shows that parts of this code are never called in tests.

I'll try to create some tests that cover these lines. I'll have to figure out what is  the `__wrapped__` for.